numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,19 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    # __new__ is overriden as a clean way to provide decorator-with-keywords
+    # syntax while allowing 'vectorize' to remain as the type of vectorized
+    # functions.
+    def __new__(cls, pyfunc=None, *args, **kw):
```

Comment:
I suppose `np._NoValue` plays the role of "sentinel that isn't `None`", but I'm curious, what's the advantage here?  Since None isn't callable